examples/ASDKgram/Sample/PhotoFeedModel.m

```diff
@@ -0,0 +1,238 @@
+  NSString       *_urlString;
+  NSUInteger     _currentPage;
+  NSUInteger     _totalPages;
+  NSUInteger     _totalItems;
+  BOOL           _fetchPageInProgress;
+  BOOL           _refreshFeedInProgress;
+  NSURLSessionDataTask *_task;
+
+  CLLocationCoordinate2D _location;
+  NSUInteger    _locationRadius;
+  NSUInteger    _userID;
+}
+
+#pragma mark - Properties
+
+- (NSMutableArray *)photos
+{
+  return _photos;
+}
+
+#pragma mark - Lifecycle
+
+- (instancetype)initWithPhotoFeedModelType:(PhotoFeedModelType)type imageSize:(CGSize)size
+{
+  self = [super init];
+  
+  if (self) {
+    _feedType    = type;
+    _imageSize   = size;
+    _photos      = [[NSMutableArray alloc] init];
+    _ids         = [[NSMutableArray alloc] init];
+    _currentPage = 0;
+    
+    NSString *apiEndpointString;
+    switch (type) {
+      case (PhotoFeedModelTypePopular):
+        apiEndpointString = fiveHundredPX_ENDPOINT_POPULAR;
+        break;
+        
+      case (PhotoFeedModelTypeLocation):
+        apiEndpointString = fiveHundredPX_ENDPOINT_SEARCH;
+        break;
+        
+      case (PhotoFeedModelTypeUserPhotos):
+        apiEndpointString = fiveHundredPX_ENDPOINT_USER;
+        break;
+        
+      default:
+        break;
+    }
+    _urlString = [[fiveHundredPX_ENDPOINT_HOST stringByAppendingString:apiEndpointString] stringByAppendingString:fiveHundredPX_CONSUMER_KEY_PARAM];
+  }
+  
+  return self;
+}
+
+#pragma mark - Instance Methods
+
+- (NSUInteger)totalNumberOfPhotos
+{
+  return _totalItems;
+}
+
+- (NSUInteger)numberOfItemsInFeed
+{
+  return [_photos count];
+}
+
+- (PhotoModel *)objectAtIndex:(NSUInteger)index
+{
+  return [_photos objectAtIndex:index];
+}
+
+- (NSInteger)indexOfPhotoModel:(PhotoModel *)photoModel
+{
+  return [_photos indexOfObjectIdenticalTo:photoModel];
+}
+
+- (void)updatePhotoFeedModelTypeLocationCoordinates:(CLLocationCoordinate2D)coordinate radiusInMiles:(NSUInteger)radius;
+{
+  _location = coordinate;
+  _locationRadius = radius;
+  NSString *locationString = [NSString stringWithFormat:@"%f,%f,%lumi", coordinate.latitude, coordinate.longitude, (unsigned long)radius];
+  
+  _urlString = [fiveHundredPX_ENDPOINT_HOST stringByAppendingString:fiveHundredPX_ENDPOINT_SEARCH];
+  _urlString = [[_urlString stringByAppendingString:locationString] stringByAppendingString:fiveHundredPX_CONSUMER_KEY_PARAM];
+}
+
+- (void)updatePhotoFeedModelTypeUserId:(NSUInteger)userID
+{
+  _userID = userID;
+  
+  NSString *userString = [NSString stringWithFormat:@"%lu", (long)userID];
+  _urlString = [fiveHundredPX_ENDPOINT_HOST stringByAppendingString:fiveHundredPX_ENDPOINT_USER];
+  _urlString = [[_urlString stringByAppendingString:userString] stringByAppendingString:@"&sort=created_at&image_size=3&include_store=store_download&include_states=voted"];
+  _urlString = [_urlString stringByAppendingString:fiveHundredPX_CONSUMER_KEY_PARAM];
+}
+
+- (void)clearFeed
+{
+  _photos = [[NSMutableArray alloc] init];
```

Comment:
We should take a closer look here in a followup - an @synchronized like you have below may be necessary here and possibly a couple other critical sections as well, like requestPageWithCompletionBlock: